oshi-core/src/main/java/oshi/hardware/platform/unix/solaris/SolarisCentralProcessor.java

```diff
@@ -52,6 +52,16 @@
     private static final long serialVersionUID = 1L;
 
     private static final Logger LOG = LoggerFactory.getLogger(SolarisCentralProcessor.class);
+    
+    private static final long BOOTTIME;
+    static {
+    	Kstat ksp = KstatUtil.kstatLookup("unix", 0, "system_misc");
+        if (ksp != null && KstatUtil.kstatRead(ksp)) {
+            BOOTTIME = KstatUtil.kstatDataLookupLong(ksp, "boot_time");
+        } else {
+        	BOOTTIME = System.currentTimeMillis / 1000L -  ksp.ks_snaptime / 1000000000L;
```

Comment:
This would work, but better to not rely on `ksp` being valid, and just use the existing `getSystemUptime()` method which will call a different kstat.

Also, I note different indentation here... did you use "Source > Format" after making these changes?

Reply:
I was concerned the getSystemUptime() method wouldn't be available without an instance of the class but on second thought that doesn't seem right. I'm refactored to use the getSystemUptime() method

Reply:
@dbwiddis I apologize for spamming you. I just have a few comments/questions. 

- First off -- It seems installing maven locally has triggered something in Eclipse because now I'm seeing helpful error markers in Eclipse where I wasn't before (which is good!).

- Second off -- These changes include now seeing the difference in indention (8 spaces instead of 4) that I couldn't see locally but were visible on your end (also good!). I will match my formatting to the source code (i.e. tab = 4 spaces).

- Third -- When trying to use getSystemUptime(), Eclipse is now complaining the way I thought it would above, namely that I "cannot make a static reference to the non-static method getSystemUptime() from the type Solaris Central Processor" . . . the two options I see to resolve this error are to make getSystemUptime() a static method or to create an instance of SolarisCentralProcessor. The second option doesn't seem right to me; not sure about the first.

- Fourth -- I did install pmd and checkstyle on eclipse but didn't realize they would find their way into the repo (wasn't my intent to include them), so I deleted .pmd and .checkstyle. That said, I submitted a PR request earlier (which I have since deleted) and deleting those files caused an issue. The project was unable to compile almost right away because the compiler was searching for the .pmd and .checkstyle files. . . which leads me to believe that by installing pmd and checkstyle on Eclipse a call to .pmd and .checkstyle was injected somewhere else in the project code -- do you know where I could find that code so I can possibly remove it?

Finally, thank you thank you for your assistance. I'm truly learning here as I go, and I really appreciate your time and patience

Reply:
> I was concerned the getSystemUptime() method wouldn't be available without an instance of the class 

Actually you are correct here!  So we can't call it from a static method.  We could make a private static method, and have the static boot time calculation and instance uptime method call it.  Alternately we could allow the boot time to be an instance-level constant, or even a normal class field calculated with a lazy getter.

Reply:
Regarding the spacing, don't use any tabs. OSHI's formatting is the "Sun Java Conventions" with that one change (tabs = 4 spaces) but there should be no tabs in submitted code.

I think you're on the right track installing maven locally to integrate with Eclipse... glad you're learning as you go, and thanks for all the effort you're putting into this!

It's okay to have those .pmd and .checkstyle files locally, just mark them in eclipse to "remove from version control". (You'll need to delete, commit, then re-add/install and then remove from version control before the next commit.)